src/test/java/com/bobocode/petros/scaner/AnnotationDependencyClassScannerTest.java

```diff
@@ -0,0 +1,79 @@
+
+    private AnnotationDependencyClassScanner sut;
+
+    @BeforeEach
+    @SneakyThrows
+    @SuppressWarnings("all")
+    void setup() {
+        sut = new AnnotationDependencyClassScanner();
+        var sutClass = sut.getClass();
+        var targetClassesPathField = sutClass.getDeclaredField("targetClassesPath");
+        targetClassesPathField.setAccessible(true);
+        targetClassesPathField.set(sut, TEST_TARGET_PATH);
+    }
+
+    @Test
+    public void ifScanSuccessfulThenValidDependencyDefinitions() {
+        var classMap = sut.scan(TEST_PACKAGE);
+        var dependencyDefinitions = classMap.get(PATH_TO_CLASS);
+        var dependencyDefinition = dependencyDefinitions.get(0);
+        var internCollectionDefinition = dependencyDefinition.getDependencyDefinitions();
+        var configClassDependency = dependencyDefinition.isConfigClassDependency();
+        var definitionName = dependencyDefinition.getName();
+        var qualifiedName = dependencyDefinition.getQualifiedName();
+        var parametersQualifiedName = internCollectionDefinition.stream()
+                .map(DependencyDefinition::getQualifiedName)
+                .collect(Collectors.toList());
+
+        assertAll(
+                () -> assertEquals(1, dependencyDefinitions.size()),
+                () -> assertEquals(2, internCollectionDefinition.size()),
+                () -> assertTrue(internCollectionDefinition.stream().anyMatch(a -> a.getName().equals("testName"))),
+                () -> assertTrue(internCollectionDefinition.stream().anyMatch(a -> a.getName().equals("listStringTest"))),
+                () -> assertEquals("java.lang.String", parametersQualifiedName.get(0)),
+                () -> assertEquals("java.util.ArrayList",parametersQualifiedName.get(1)),
+                () -> assertTrue(classMap.containsKey(PATH_TO_CLASS)),
+                () -> assertFalse(configClassDependency),
+                () -> assertEquals(DEFINITION_NAME, definitionName),
+                () -> assertEquals(QUALIFIED_NAME, qualifiedName),
+                () -> assertDoesNotThrow(() -> new DefaultConstructorNotFoundException(QUALIFIED_NAME)),
+                () -> assertDoesNotThrow(() -> new MultipleInjectConstructorsException(QUALIFIED_NAME)),
+                () -> assertDoesNotThrow(() -> new ClassNotFoundException(QUALIFIED_NAME)),
+                () -> assertDoesNotThrow(() -> new NoSuchPackageFoundException(PATH_TO_CLASS))
+        );
+    }
+
+    @Test
+    void ifInvalidPackageNameThenThrowNoSuchPackageFoundException() {
+        var invalidPackageName = "invalid.package";
+        var noSuchPackageFoundException = assertThrows(NoSuchPackageFoundException.class, () -> sut.scan(invalidPackageName));
+        var expectedMessage = String.format("No package with name = %s found", invalidPackageName);
+        assertEquals(expectedMessage, noSuchPackageFoundException.getMessage());
+    }
+}
```

Comment:
Not important in our case, just pay attention on this warning. Just FYI - https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline